src/Microsoft.AspNet.Mvc.Razor/Compilation/ExpressionRewriter.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNet.Mvc.Razor.Compilation
+{
+    /// <summary>
+    /// An expression rewriter which can hoist a simple expression lambda into a private field.
+    /// </summary>
+    public class ExpressionRewriter : CSharpSyntaxRewriter
+    {
+        private static readonly string FieldNameTemplate = "__h{0}";
```

Comment:
Absolutely. You won't see an artifact like that if the expression is on a single line because the debugger highlights the line.

If we need to address it we can increment 'h' as well.